common/lib/libutil.c

```diff
@@ -143,3 +143,19 @@ void convert_uint8_t_pointer_to_uint32_t(uint32_t *data_32, uint8_t *data_8, uin
 		*data_32 = (data_8[0] << 24) + (data_8[1] << 16) + (data_8[2] << 8) + data_8[3];
 	}
 }
+
+double power(double x, double y)
```

Comment:
`y` should be type of `int` instead of `double` as the actual algorithm won't work with non int exponent, or at least cast to int to approximate.

Reply:
Got it, thanks. Please have a look at new commit.